packages/plugins/sentry/server/middlewares/sentry-performance.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { stripUrlQueryAndFragment } = require('@sentry/utils');
+
+/**
+ * Programmatic sentry-performance middleware. We do not want to expose it in the plugin
+ * @param {{ strapi: import('@strapi/strapi').Strapi }}
+ */
+module.exports = ({ strapi }) => {
+  const sentryService = strapi.plugin('sentry').service('sentry');
+  sentryService.init();
+  const Sentry = sentryService.getInstance();
+
+  if (!Sentry || !sentryService.hasTracingEnabled()) {
+    // initialization failed or tracing is not enabled
+    return;
+  }
+
+  strapi.server.use((ctx, next) => {
+    Sentry.runWithAsyncContext(async () => {
+      const hub = Sentry.getCurrentHub();
+      hub.configureScope((scope) =>
+        scope.addEventProcessor((event) =>
+          Sentry.addRequestDataToEvent(event, ctx.request, {
+            include: {
+              user: false,
+            },
+          })
+        )
+      );
+
+      next();
+    });
+  });
+
+  // this tracing middleware creates a transaction per request
+  strapi.server.use((ctx, next) => {
+    const reqMethod = (ctx.method || '').toUpperCase();
+    const reqUrl = ctx.url && stripUrlQueryAndFragment(ctx.url);
+
+    // connect to trace of upstream app
+    let traceparentData;
+    if (ctx.request.get('sentry-trace')) {
+      traceparentData = Sentry.extractTraceparentData(ctx.request.get('sentry-trace'));
```

Comment:
we also want to extract baggage here, and also use distributed tracing features (don't gate by `hasTracingEnabled`). See express as an example: 

https://github.com/getsentry/sentry-javascript/blob/53d32a3ac7d2238616ac6dc9c210e3d1d0584bbb/packages/node/src/handlers.ts#L56-L69

Reply:
I updated this again, can you have another look?